gadgets/trace_capabilities/test/unit/trace_capabilities_test.go

```diff
@@ -0,0 +1,195 @@
+package tests
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/unix"
+
+	gadgettesting "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
+	utilstest "github.com/inspektor-gadget/inspektor-gadget/internal/test"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/gadgetrunner"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/utils"
+)
+
+type ExpectedTraceCapEvent struct {
+	Proc utils.Process `json:"proc"`
+
+	Cap     int32  `json:"cap"`
+	Syscall string `json:"syscall"`
+	Audit   int32  `json:"audit"`
+	Verdict int32  `json:"verdict"`
+}
+
+type testDef struct {
+	runnerConfig   *utilstest.RunnerConfig
+	mntnsFilterMap func(info *utilstest.RunnerInfo) *ebpf.Map
+	generateEvent  func() (int, error)
```

Comment:
If `generateCapabilityEvent` is made generic as suggested in the other comment, this can be removed and instead this can take the capability number to be used.